### DISCUSSION TITLE
拖拽节点生成的画布，如何获取其相关节点数据呢，处理一下传给后端

### DISCUSSION THREAD

**Discussion by xiazhaohui** [antvis/X6#2042] [Q&A] [answered]
需求是做一个自动化有营销流程画布，自定义流程走向和发生一些动作。
画布左侧是一些自定义的图形节点，手动拖拽它们组成关系结构。怎样去拿到拖拽形成的结构数据呢，比如某节点的value值和与其他节点的关系。
![飞书20220422-171948](https://user-images.githubusercontent.com/82207908/164676853-b6996468-0be8-4082-9d70-14c39248964d.png)

**Comment by NewByVector** [comment, marked as answer]
画布上所有元素的数据可以通过以下方法拿到：
```ts
graph.toJSON()
```
单个节点的业务数据，可以放在 data 属性中
```ts
graph.setData(daa)
```

**Reply by xiazhaohui** [reply]
懂了，感谢感恩
